### DISCUSSION TITLE
Tesla WiFi Dropouts

### DISCUSSION THREAD

**Discussion by Dazzert318** [tesla-android/issue-tracker#38] [Q&A]
All installed and deployed to vehicle which is a 2022 Model 3. I’ve tried using the system on 3 short trips of around 30-40 minute duration with Waze. On each occasion Tesla WiFi is disconnecting randomly from Tesla-Android and at that point the browser locks up. It looks like the WiFi on the Model 3 is turning off or going to sleep and when I try to reconnect the Tesla-Android will only re-establish a connection once I’ve rebooted the two RPi units. Does anyone have a fix or advice?

**Comment by virtuality222** [comment]
This is not an answer and not sure how to just add a normal response (sorry new to Github) - but wanted to say I have this exact same issue, only solved in the same way as well. Although it has happened that the only way to solve is to ALSO reboot the Tesla itself (hold down both buttons on the steering wheel for 20 seconds). Maybe 3-4 times so far.
I have a 2021 Model 3.

**Comment by ochompsky** [comment]
the tesla unfortunately has a bad wifi antenna and buggy wifi code in general.  don't think there is a permanent fix.  i think the best bet may be to plug in a stronger external usb wifi card into the pi and use that instead.

**Reply by virtuality222** [reply]
Any chance for a recommendation on a stronger USB Wifi card?

**Reply by ochompsky** [reply]
> Any chance for a recommendation on a stronger USB Wifi card?

not yet, i haven't really had the project working long enough to determine if i need one or not.  or even if im gonna keep it as is.

**Comment by rrlevy** [comment]
I noticed that this behavior happens when the LTE Modem loses internet connectivity.

When this happens, the car automatically disconnects from the Wifi netwotk and therefore the video streaming stops....

I don't know if there is any way to fix that, unless @mikegapinski finds a way to "trick" the car that the Internet is always on by some Pi Hole configuration

When using the car in remote areas that lacks good LTE coverage, this issue renders the whole Tesla Android system unusable during those trips.

**Reply by ochompsky** [reply]
maybe for you but the flaky wifi in the car is far more likely to be the culprit.  tesla doesn't disconnect immediately if you lose internet.  i've had it work for up to 30 minutes with a disconnected LTE modem.

**Reply by rrlevy** [reply]
Really? I was only having Wifi dropouts when I noticed my iPhone was losing connection too. The Tesla Android Wifi signal has always been strong in my Model Y

**Reply by ochompsky** [reply]
maybe the antenna is better on the Y, it's pretty notoriously bad in the 3

**Comment by BroRidersMTL** [comment]
I think the issue is the LTE modem when hidden is less efficient maybe an extended cable to keep modem visible

**Reply by rrlevy** [reply]
If the area you are in has poor connection, even making the modem visible will not solve it....
